plugins/samples/jwt_auth/plugin.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_jwt_auth]
+#include <boost/url/parse.hpp>
+#include <boost/url/url.hpp>
+
+#include "jwt_verify_lib/verify.h"
+#include "proxy_wasm_intrinsics.h"
+
+class MyRootContext : public RootContext {
+ public:
+  explicit MyRootContext(uint32_t id, std::string_view root_id)
+      : RootContext(id, root_id) {}
+
+  bool onConfigure(size_t config_len) override {
+    // Fetch plugin config and take ownership of the buffer.
+    config_ =
+        getBufferBytes(WasmBufferType::PluginConfiguration, 0, config_len);
+
+    // Read the RSA key from the config file.
+    const auto rsa_key = config_->toString();
+    jwks = google::jwt_verify::Jwks::createFrom(
+        rsa_key, google::jwt_verify::Jwks::Type::PEM);
+    return jwks->getStatus() == google::jwt_verify::Status::Ok;
+  }
+
+  google::jwt_verify::JwksPtr jwks;
```

Comment:
Suggest making the field private, and either defining an accessor method, or sharing a pointer to it with MyHttpContext via an added MyHttpContext constructor param.

Reply:
Done